CMakeLists.txt

```
@@ -351,7 +351,7 @@ if(CMAKE_BUILD_TYPE STREQUAL "Check" AND TEST_CPP)
 endif()
 
 if (NOT EXISTS "${MBEDTLS_FRAMEWORK_DIR}/CMakeLists.txt")
-    if (EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/.git/")
+    if (EXISTS "${$MBEDTLS_DIR}/.git/")
```

Even with this change, I always get the “does not appear to be a git checkout” message from CMake, both in-tree and out-of-tree.

```
% git status
Not currently on any branch.
Changes not staged for commit:
  (use "git add/rm <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        deleted:    framework

Untracked files:
  (use "git add <file>..." to include in what will be committed)
        not.framework/

no changes added to commit (use "git add" and/or "git commit -a")
% cmake .
Traceback (most recent call last):
  File "/tmp/9833-fix-submodule-error-messages-development/scripts/config.py", line 19, in <module>
    from mbedtls_framework import config_common
ModuleNotFoundError: No module named 'mbedtls_framework'
CMake Error at CMakeLists.txt:357 (message):
  
  /tmp/9833-fix-submodule-error-messages-development/framework/CMakeLists.txt
  not found (and does not appear to be a git checkout).  Please ensure you
  have downloaded the right archive from the release page on GitHub.


-- Configuring incomplete, errors occurred!
```

---

There's an extra `$`. Also, it should be `.git` without a trailing slash, because in a worktree, `.git` is a text file, not a directory.

```suggestion
    if (EXISTS "${MBEDTLS_DIR}/.git")
```

---

Fixed both, it appears to work correctly now.